commodore/cluster.py

```diff
@@ -19,89 +23,105 @@ def fetch_cluster(cfg, clusterid):
-
-    cluster_distro = cluster_facts['distribution']
-    cloud_provider = cluster_facts['cloud']
-    cluster_id = cluster['id']
-    tenant = cluster['tenant']
-    component_defaults = [f"defaults.{cn}" for cn in components if
-                          (P('inventory/classes/defaults') / f"{cn}.yml").is_file()]
-    global_defaults = ['global.common',
-                       f"global.distribution.{cluster_distro}",
-                       f"global.cloud.{cloud_provider}"]
-    if 'region' in cluster_facts and cluster_facts['region']:
-        global_defaults.append(f"global.cloud.{cloud_provider}.{cluster_facts['region']}")
-
-    if 'lieutenant-instance' in cluster_facts and cluster_facts['lieutenant-instance']:
-        global_defaults.append(
-            f"global.lieutenant-instance.{cluster_facts['lieutenant-instance']}")
-    global_defaults.append(f"{tenant}.{cluster_id}")
-    commodore_facts = {
-        'target_name': 'cluster',
-        'cluster': {
-            'name': cluster_id,
-            'catalog_url': catalog,
-            'tenant': tenant,
-            # TODO Remove dist after deprecation phase.
-            'dist': cluster_distro,
-        },
-        # TODO Remove the facts below after deprecation phase.
-        'cloud': {
-            'provider': cloud_provider,
-        },
-        'customer': {
-            'name': tenant,
-        },
+
+    return response
+
+
+def render_target(target: str, components: Iterable[str]):
+    classes = [f"params.{target}"]
+
+    for component in components:
+        defaults_file = P('inventory/classes/defaults') / f"{component}.yml"
+        if defaults_file.is_file():
+            classes.append(f"defaults.{component}")
+
+    # TODO Remove before merge!
+    classes.append('common')
+    classes.append('global.common')
+
+    return {
+        'classes': classes,
     }
-    # TODO Remove after deprecation phase.
-    if 'region' in cluster_facts:
-        commodore_facts['cloud']['region'] = cluster_facts['region']
-    target = {
-        'classes': component_defaults + global_defaults,
+
+
+def target_file(target: str):
+    return P('inventory/targets') / f"{target}.yml"
+
+
+def update_target(cfg: Config, target):
+    click.secho('Updating Kapitan target...', bold=True)
+    file = target_file(target)
+    os.makedirs(file.parent, exist_ok=True)
+    yaml_dump(render_target(target, cfg.get_components().keys()), file)
+
+    # TODO Remove before merge!
+    common = P('inventory/classes/common.yml')
+    os.makedirs(common.parent, exist_ok=True)
+    yaml_dump({
+        'classes': [
+            'global.distribution.${facts:distribution}',
+            'global.cloud.${facts:cloud}',
+            'global.cloud.${facts:cloud}.${facts:region}',
+            '${cluster:tenant}.${cluster:name}',
+
+        ]}, common)
+
+
+def render_params(cluster, target: str):
+    facts = cluster['facts']
+    for fact in ['distribution', 'cloud']:
```

Comment:
Is this still something we should enforce?
I'm just thinking how generic these really are. We don't have any dependency on hard coded facts anymore anywhere else...

Reply:
You are right. This is no longer needed. I will remove it.

Reply:
Um wait! We need to keep it until we remove the deprecated `cluster.dist` and `cloud.provider` parameters.